PyPDF2/_page.py

```diff
@@ -1069,48 +1075,188 @@ def extract_text(self, Tj_sep: str = "", TJ_sep: str = "") -> str:
 
         :return: a string object.
         """
-        text = ""
+        # code freely inspired from @twiggy ; see #711
+        def buildCharMap(font_name : str)->Tuple[str,Dict,Dict]:
+            mapDict : Any = {}
+            processRg : bool = False
+            processChar : bool = False
+            encoding : List[str] = []
+            fontType :str = cast(NameObject, cast(DictionaryObject,
+                                 cast(DictionaryObject,
+                                      cast(DictionaryObject, self["/Resources"])["/Font"]
+                                 )[font_name]
+                            )["/Subtype"])
```

Comment:
Cast is telling mypy to ignore what it thinks the type is. Typically you only need one cast per line (if you need any at all)

```suggestion
            fontType :str = cast(NameObject, self["/Resources"]["/Font"][font_name]["/Subtype"])
```